server-plugins/activity-resources/src/index.ts

```diff
@@ -263,10 +263,29 @@ async function ActivityMessagesHandler (tx: TxCUD<Doc>, control: TriggerControl)
   return await generateDocUpdateMessages(tx, control)
 }
 
+async function OnDocRemoved (originTx: TxCUD<Doc>, control: TriggerControl): Promise<Tx[]> {
+  const tx = TxProcessor.extractTx(originTx) as TxCUD<Doc>
+
+  if (tx._class !== core.class.TxRemoveDoc) {
+    return []
+  }
+
+  const activityDocMixin = control.hierarchy.classHierarchyMixin(tx.objectClass, activity.mixin.ActivityDoc)
+
+  if (activityDocMixin === undefined) {
+    return []
+  }
+
+  const messages = await control.findAll(activity.class.ActivityMessage, { attachedTo: tx.objectId })
```

Comment:
Please use projection, since we do not really need documents.

Reply:
Done.